src/main/java/spoon/reflect/reference/CtTypeReference.java

```diff
@@ -50,13 +50,15 @@
 	/**
 	 * Gets the Java runtime class of the referenced type.
 	 *
-	 * This is a low-level feature, it should never been used.
-	 * For CtTypeReference, use {@link #getTypeDeclaration()} instead,
+	 * This is a low-level feature, it should never been used, use {@link #getTypeDeclaration()} instead
 	 * in order to only stay in the Spoon world and manipulate CtType instead of java.lang.Class.
+	 * Note:
+	 * - it will return null if the class is not the classpath
+	 * - it will fail if the static initializers of the class fail
+	 * - the Class object may not reflect the CtType if case you made recent changes to it, only the version from the classpath is taken into account
```

Comment:
```suggestion
	 * <br/>
	 * <br/>
	 * Note (these are artifacts of the <em>current implementation</em> and might change at any time):
	 * <ol>
	 *     <li>
	 *         it will return the version of the class your {@link spoon.compiler.Environment#getInputClassLoader()}
	 *         loads. This might be a version spoon compiled
	 *         (manually using {@link spoon.SpoonModelBuilder#compile(spoon.SpoonModelBuilder.InputType...)}
	 *     	   or implicitly using {@link spoon.compiler.Environment#shouldCompile})
	 *     	   or from your classpath (which might be null)
	 *     </li>
	 *     <li>it will run static initializers (executing arbitrary code from the analyzed project in the process)</li>
	 *     <li>it will fail if the static initializers of the class fail</li>
	 *     <li>the Class object may not reflect the CtType if case you made recent changes to it based on point 1</li>
	 * </ol>
```